docs/reference/modules/cross-cluster-search.asciidoc

```diff
@@ -372,3 +372,42 @@ image:images/ccs/ccs-dont-min-roundtrip-shard-results.svg[]
 final results in the {ccs} response.
 +
 image:images/ccs/ccs-min-roundtrip-client-response.svg[]
+
+[float]
+[[ccs-supported-configurations]]
+==== Supported configurations
+
+<<gateway-nodes-selection, Remote clusters>> shows which versions can talk
+to which other versions but there are at least three nodes involved The
+"coordinating node" receives your search request and sends it to a
+"gateway node" in each cluster. The "gateway nodes" in turn send your search
+request to "data nodes" that have a copy of the shards.
+
+During a <<rolling-upgrades, rolling upgrade>> all three nodes could be running
+a different version of Elasticsearch! This is supported *but* it gets less
+testing than the more common scenarios where the remote cluster is on the same
+version. Everything should work either way, but don't tempt fate, ok?
```

Comment:
I wonder if we should rather start with talking about clusters rather than nodes, and mostly leave gateway nodes out to begin with. I think talking about nodes may tempt users to have different versions in the same cluster on purpose to use CCS in some unsupported way. I would start describing the scenario that we supported with three versions, where the middle one is the CCS cluster that may or may not have data. Then I would expand talking about edge cases, meaning rolling upgrades hence different versions within the same cluster, hence gateway nodes and what may or may not happen.

Reply:
I think I understand what you are asking for. I'll give it a shot!

Reply:
+1, I like the idea of describing in terms of clusters (instead of nodes) too, easier to understand.